include/swift/SIL/SILCloner.h

```diff
@@ -270,6 +270,7 @@ class SILBuilderWithPostProcess : public SILBuilder {
     {
       setInsertionPoint(SC.getBuilder().getInsertionBB(),
                         SC.getBuilder().getInsertionPoint());
+      setOpenedArchetypesTracker(SC.getBuilder().getOpenedArchetypesTracker());
```

Comment:
@eeckstein Can you submit this fix a separate commit? I think we may also take it into the swift-3.0-branch.